wire-ios-data-model/Tests/UseCases/AccountImageSource/GetTeamAccountImageSourceUseCaseTests.swift

```diff
@@ -0,0 +1,162 @@
+            userContext: user.managedObjectContext,
+            account: coreDataStack.account
+        )
+
+        // Then
+        guard case .image(let teamImage) = teamImageSource, teamImage.pngData() == teamImageData else {
+            return XCTFail("Expected team image to match actual image")
+        }
+    }
+
+    func testTeamName() async throws {
+        // Given
+        let user = await coreDataStack.viewContext.perform { [self] in
+            let user = ZMUser.selfUser(in: coreDataStack.viewContext)
+            let team = Team.mockTeam(context: coreDataStack.viewContext)
+            team.name = "\tWire \n"
+            let membership = TeamMembership.insertNewObject(in: coreDataStack.viewContext)
+            membership.team = team
+            membership.user = user
+            return user
+        }
+
+        // When
+        let teamImageSource = try await sut.invoke(
+            user: user,
+            userContext: user.managedObjectContext,
+            account: coreDataStack.account
+        )
+
+        // Then
+        XCTAssertEqual(teamImageSource, .text("W"))
+    }
+
+    func testAccountImage() async throws {
+        // Given
+        let teamImageData = try imageData(from: .brown)
+        coreDataStack.account.teamImageData = teamImageData
+        let user = await coreDataStack.viewContext.perform { [self] in
+            ZMUser.selfUser(in: coreDataStack.viewContext)
+        }
+
+        // When
+        let teamImageSource = try await sut.invoke(
+            user: user,
+            userContext: nil,
+            account: coreDataStack.account
+        )
+
+        // Then
+        guard case .image(let teamImage) = teamImageSource, teamImage.pngData() == teamImageData else {
+            return XCTFail("Expected team image to match actual image")
+        }
+    }
+
+    func testAccountName() async throws {
+        // Given
+        coreDataStack.account.teamName = "\nWire \t"
+        let user = await coreDataStack.viewContext.perform { [self] in
+            ZMUser.selfUser(in: coreDataStack.viewContext)
+        }
+
+        // When
+        let teamImageSource = try await sut.invoke(
+            user: user,
+            userContext: user.managedObjectContext,
+            account: coreDataStack.account
+        )
+
+        // Then
+        XCTAssertEqual(teamImageSource, .text("W"))
+    }
+
+    func testNoSource() async throws {
+        // Given
+        let user = await coreDataStack.viewContext.perform { [self] in
+            ZMUser.selfUser(in: coreDataStack.viewContext)
+        }
+
+        do {
+            // When
+            let teamImageSource = try await sut.invoke(
+                user: user,
+                userContext: user.managedObjectContext,
+                account: coreDataStack.account
+            )
+            XCTFail("Unexpected success")
+        } catch GetTeamAccountImageSourceUseCase.Error.invalidImageSource {
+            // Then
+        }
```

Comment:
```suggestion
        } catch {
            XCTFail("Unexpected error: \(error)")
        } 
```

Optional: Consider failing on default error case ()

Reply:
The test will fail in that case, do you still prefer to write it explicitly?